tests/utils/couchdb-utils.js

```diff
@@ -0,0 +1,42 @@
+import PouchDb from 'pouchdb-core';
+import http from 'pouchdb-adapter-http';
+import { promises as fs } from 'node:fs';
+import path from 'path';
+
+PouchDb.plugin(http);
+
+const env = process.env;
+
+export const dbNames = env.COUCHDB_DBS.split(',');
+const dbUrl = name => `http://127.0.0.1:${env.COUCHDB_PORT}/${name}`;
+
+export const docs = [];
+
+const loadDocs = async () => {
+  const dir = path.join(import.meta.dirname, '..', 'data', 'json_docs');
+  for (const file of await fs.readdir(dir)) {
+    const data = await fs.readFile(path.join(dir, file), 'utf-8');
+    docs.push(JSON.parse(data));
+  }
+};
+
+const importDocs = async (dbName) => {
+  const opts =  { auth: { username: env.COUCHDB_USER, password: env.COUCHDB_PASSWORD, skip_setup: false } };
+  const db = new PouchDb(dbUrl(dbName), opts);
+
+  const dbDocs = docs.map(doc => ({  ...doc, _id: `${dbName}-${doc._id}` }));
```

Comment:
`${dbName}-${doc._id}` instead of `${doc._id}` adds some weirdness with references, is it really necessary?

Reply:
Yes :) 
Because otherwise both databases would have the same docs. And I want them to have "different" docs.

Reply:
not sure exactly what you mean...like the two couchdb databases, medic and medic_sentinel for example, would have the same docs? if so, its a problem for references, like in the json docs if a contact "aaa" has a parent "bbb", it ends up in the db with id "medic-bbb" but without replacing the reference, "aaa" ends up in postgres with parent id "bbb", which now doesn't exist... 

 or the "source" (the json docs) and "target" (test couchdb) database have the same docs? in which case why do they need to be different?

Reply:
Ummmm lol. welcome to how our PG works. 
All CHT databases are copied into a single PG table, this has been the case forever. 
The unique field for documents is _id, which should not have conflicts across these CHT databases (ideally - this is sort of naive but this is how people want it), mainly because they are uuids with low conflict rates, and we purposefully NEVER create a doc with the same uuid in two databases, even if two docs are "connected", their _id fields are either sufixed or prefixed with something. 

In the case of this test, I wanted to check that both databases have their docs copied to the main pg table. This means that all docs would need to have different _id fields, so I chose to prefix them. 

I don't exactly understand the source of confusion here.

Reply:
OK so its just for this test where the same document is in both databases, which is not a situation we would expect in the real world.

the problem is when expanding these tests beyond just counting the number of documents; in postgres the expectation is that if you have a document in couchdb with an _id you expect to be able to find a row the corresponding tables in postgres with that same id. Having to add a prefix is maybe ok, but adds complexity. Then if a document in couchdb refers to another document (e.g. contact with a parent), that needs to translate into a foreign key relationship. And there it breaks completely if the documents have their ids changed but references not changed.

But it is only an issue where expanding these tests, so I guess beyond the scope of this PR

Reply:
Oh, yea, you're right, when we introduce relations between these docs and want to test the **DBT model**, then the _ids are relevant. 
I agree that this would be an issue in a dbt model oriented test, but I structured these as sync copy sanity check tests. 
We can easily revisit them when or if we want to change the tests to cover models too instead of just the copying.